includes/html/api_functions.inc.php

```diff
@@ -2338,6 +2338,34 @@ function list_vlans(Illuminate\Http\Request $request)
     return api_success($vlans, 'vlans');
 }
 
+function list_ports_vlans(Illuminate\Http\Request $request)
+{
+    $sql = '';
+    $sql_params = [];
+    $hostname = $request->get('hostname');
+    $device_id = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+    if (is_numeric($device_id)) {
+        $permission = check_device_permission($device_id);
+        if ($permission !== true) {
+            return $permission;
+        }
+        $sql = ' AND `devices`.`device_id` = ?';
+        $sql_params[] = $device_id;
+    }
+    if (! Auth::user()->hasGlobalRead()) {
+        $sql .= ' AND `ports_vlans`.`device_id` IN (SELECT device_id FROM devices_perms WHERE user_id = ?)';
```

Comment:
This doesn't work with device group permissions. build this query with eloquent instead, then you can simply use scopeHasAccess()

Reply:
would that still be used in `api_functions.inc.php`?

Reply:
Yes, there are probably some queries that use it there already.